### ISSUE TITLE
Specify Minecraft version for fabric mod

### ISSUE THREAD

**Issue by kaylendog** [EngineHub/WorldEdit#2225]
### The Problem

Currently, WorldEditFabric does not specify a Minecraft version in its `fabric.mod.json`, which leads to confusion, wasted time, and avoidable crashes. Consider the latest 1.19 versions - WorldEdit for 1.19.3 is not compatible with 1.19.2, which should be obvious, but this isn't specified in the mod definition, JAR name, or anywhere else, other than the CurseForge mod release title.

### A Solution

Add the appropriate version to `fabric.mod.json` and update it when appropriate.

**Comment by octylFractal** [comment]
It is in fact _not_ obvious, minor releases have historically been compatible (e.g. 1.19.1 -> 1.19.2 was compatible, and 1.17 -> 1.17.1). We should specify `[1.19.3, 1.20)` like we do with Forge though. 

We have no control over CF's awful UI, we did mark it as 1.19.3 only in the actual metadata:

![worldedit 7.2.13 is 1.19.3](https://url.octyl.net/nx6z2orw8lcmgaiguro6)

**Comment by kaylendog** [comment]
> ~It is in fact _not_ obvious, minor releases have historically been compatible (e.g. 1.19.1 -> 1.19.2 was compatible, and 1.17 -> 1.17.1).~ (Edit: Misread the order of what you said about the version.) We should specify `[1.19.3, 1.20)` like we do with Forge though.
> 
> We have no control over CF's awful UI, we did mark it as 1.19.3 only in the actual metadata:
> 
> ![worldedit 7.2.13 is 1.19.3](https://camo.githubusercontent.com/7b835e41698e8db83508ff75db7607bb282ba21279a897236f39cf6437e7b7c7/68747470733a2f2f75726c2e6f6374796c2e6e65742f6e78367a326f7277386c636d6761696775726f36)

Yeah, I've dealt with CurseForge's release page myself, not a fan. 

Guess the `fabric.mod.json` update is really the only quality-of-use thing we can do here?

**Comment by wizjany** [comment]
would note that even `[1.19.3, 1.20)` is a guess, as a potential 1.19.4 could be incompatible. also minor releases can be partially incompatible, like 1.19.2 was with WECUI where only one button didn't work but everything else did. on one hand our policy is not to support old patch versions, but also hard-excluding older, potentially partially incompatible versions means new fixes might not be available on those versions even if they are minor, even protocol-compatible changes.

**Comment by kaylendog** [comment]
> also, fabric.mod.json isn't visible to anything beside the modloader really. it doesn't help people figure out what version to download. the best it does it turn an ugly error into a more legible error at load-time.

Was testing this just now - the game appears fine until you load into a world, so it's a little disconcerting. Fabric's loading error GUI isn't exactly the friendliest, I agree, but I'd argue it's better than trying to solve why WorldEdit is looking for classes that don't exist!

**Comment by kaylendog** [comment]
I had a [similar issue](https://github.com/shedaniel/RoughlyEnoughItems/issues/1248) with REI the other day, they solved it by dumping a bunch of logs in the console before the game printed out 100,000 lines in a few seconds. I'd suggest not doing that and going for the horrible GUI approach.

**Comment by InnocenTurbo** [comment]
At least they solved th problem now by not making it possible to download any 1.19 version other than 1.19.3. great stuff, i have no cleu what do with this but i just wanted to let people know so that it might be fixed.

**Comment by wizjany** [comment]
> At least they solved th problem now by not making it possible to download any 1.19 version other than 1.19.3. great stuff, i have no cleu what do with this but i just wanted to let people know so that it might be fixed.

not sure what you mean by that, we haven't made previous releases unavailable in any way. everything is still on curseforge with versions tagged as described previously.

**Comment by InnocenTurbo** [comment]
if i attemt to download worldedit it leads me to the download for 1.19.3, no matter what 1.19 version click on

**Comment by me4502** [comment]
> if i attemt to download worldedit it leads me to the download for 1.19.3, no matter what 1.19 version click on

That sounds like an issue on your end, we have no control over how those sites work and therefore cannot do this

**Comment by InnocenTurbo** [comment]
just wanted to make it know, i have no idea how that kinda stuff works, ty for the reply tho

**Comment by shedaniel** [comment]
> I had a [similar issue](https://github.com/shedaniel/RoughlyEnoughItems/issues/1248) with REI the other day, they solved it by dumping a bunch of logs in the console before the game printed out 100,000 lines in a few seconds. I'd suggest not doing that and going for the horrible GUI approach.

Your issue is a bug with Quilt, the log spam is not intended, Quilt Loader broke classpath access to Minecraft itself, which they fixed in a beta release and haven't released it. That isn't a REI issue.

**Comment by kaylendog** [comment]
> > I had a [similar issue](https://github.com/shedaniel/RoughlyEnoughItems/issues/1248) with REI the other day, they solved it by dumping a bunch of logs in the console before the game printed out 100,000 lines in a few seconds. I'd suggest not doing that and going for the horrible GUI approach.
> 
> Your issue is a bug with Quilt, the log spam is not intended, Quilt Loader broke classpath access to Minecraft itself, which they fixed in a beta release and haven't released it. That isn't a REI issue.

Yah, I was more referring to logging to console over using Fabric's dependency mismatch GUI.
